tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,412 @@
+
+        clone_yaml = self.build_params()
+        self.get_log_names()
+        self.run_fio_on_pod(file_size)
+
+        # Running the test
+        results = []
+        for test_num in range(1, int(self.params["clonenum"]) + 1):
+            log.info(f"Starting test number {test_num}")
+            ct = self.create_clone(test_num, clone_yaml)
+            speed = self.params["datasize"] / ct
+            results.append({"Clone Num": test_num, "time": ct, "speed": speed})
+            log.info(
+                f"Results for clone number {test_num} are : "
+                f"Creation time is {ct} secs, Creation speed {speed} MB/sec"
+            )
+
+        for r in results:
+            log.info(
+                f"Clone number {r['Clone Num']} creation time is {r['time']} secs."
+            )
+            log.info(
+                f"Clone number {r['Clone Num']} creation speed is {r['speed']} MB/sec."
+            )
+
+    def build_params(self):
+        log.info("Start building params")
+
+        self.params["nspace"] = self.pvc_obj.namespace
+        self.params["pvcname"] = self.pvc_obj.name
+
+        log_file_name = os.path.basename(__file__).replace(".py", ".log")
+
+        full_log = f"{ocsci_log_path()}/{log_file_name}"
+        logging.basicConfig(
+            filename=full_log, level=logging.INFO, format=constants.LOG_FORMAT
+        )
+
+        self.params["datasize"] = int(self.params["filesize"].replace("M", ""))
+
+        self.params["clone_yaml"] = constants.CSI_CEPHFS_PVC_CLONE_YAML
+        if self.interface == constants.CEPHBLOCKPOOL:
+            self.params["clone_yaml"] = constants.CSI_RBD_PVC_CLONE_YAML
+
+        output = self.run_oc_command(
+            cmd=f"get pod {self.pod_obj.name} -o yaml", namespace=self.params["nspace"]
+        )
+
+        results = yaml.safe_load("\n".join(output))
+        self.params["path"] = results["spec"]["containers"][0]["volumeMounts"][0][
+            "mountPath"
+        ]
+        log.info(f"path - {self.params['path']}")
+
+        fd, tmpfile = tempfile.mkstemp(suffix=".yaml", prefix="Clone")
+        self.params["tmpfile"] = tmpfile
+
+        # reading template of clone yaml file
+        with open(self.params["clone_yaml"], "r") as stream:
+            try:
+                clone_yaml = yaml.safe_load(stream)
+                clone_yaml["spec"]["storageClassName"] = self.pvc_obj.backed_sc
+                clone_yaml["spec"]["dataSource"]["name"] = self.params["pvcname"]
+                clone_yaml["spec"]["resources"]["requests"]["storage"] = (
+                    str(self.pvc_obj.size) + "Gi"
+                )
+            except yaml.YAMLError as exc:
+                log.error(f"Can not read template yaml file {exc}")
+        log.info(
+            f'Clone yaml file : {self.params["clone_yaml"]} '
+            f"Content of clone yaml file {clone_yaml}"
+        )
+
+        return clone_yaml
+
+    def get_log_names(self):
```

Comment:
Is this function being used somewhere in the test?

Reply:
@keesturam yes, this function is used to get log names that are stored in params dictionary. This dicrionary is later used to read creation times and calculate creation time period.